src/main/java/seedu/address/logic/commands/EditTaskCommand.java

```diff
@@ -46,6 +47,8 @@
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
     public static final String MESSAGE_NOTHING_TO_EDIT = "At least one field need to be different to edit.";
     public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    public static final String MESSAGE_DRIVER_DURATION_FIELD_CONSTRAINTS =
+            "Driver AND Duration field are compulsory IF driver was not assigned initially.";
```

Comment:
I feel that we should route all requests to change `Duration` and `Driver` via the `assign` command, because a task and a driver's schedule is highly coupled. So that we can 

1. avoid complex null checks and condition checks that could very potentially cause bugs
2. increase code reuse 
3. user can get notified of a better timeslot when editing the task too

So `edit` is only for non-Task related items, like `Description` and reassigning customers. If the manager want to change the driver or time for any task, one possible workflow is that, he/she will first 1) unassign the existing driver 2) use `assign` to specify his/her desired changes. 

However, this is still up to discussion. I just feel such workflow is more streamlined, and the user gets the nice timeslot suggestion feature.

Reply:
Lets discuss this